internal/utils/chaincode.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package utils
+
+import (
+	"github.com/golang/protobuf/proto"
+	"github.com/hyperledger/fabric-chaincode-go/shim"
+	"github.com/hyperledger/fabric-protos-go/peer/lifecycle"
+)
+
+func GetChaincodeDefinition(chaincodeId string, stub shim.ChaincodeStubInterface) (*lifecycle.QueryApprovedChaincodeDefinitionResult, error) {
+	function := "QueryApprovedChaincodeDefinition"
+	args := &lifecycle.QueryApprovedChaincodeDefinitionArgs{
+		Name:     chaincodeId,
+		Sequence: 0, // TODO get current sequence number
+	}
+	argsBytes, err := proto.Marshal(args)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := stub.InvokeChaincode("_lifecycle", [][]byte{[]byte(function), argsBytes}, stub.GetChannelID())
+
+	df := &lifecycle.QueryApprovedChaincodeDefinitionResult{}
+	if err := proto.Unmarshal(resp.Payload, df); err != nil {
+		return nil, err
+	}
+	return df, nil
+}
+
+func ExtractMrEnclaveFromChaincodeDefinition(chaincodeId string, stub shim.ChaincodeStubInterface) (string, error) {
```

Comment:
" ExtractMrEnclaveFromChaincodeDefinition" seems a rather verbose name hidding a bit the key part. what about a simple "GetMrEnclave" or "GetRegisteredMrEnclave"?